### PR TITLE
Dan CI guides

### PR DESCRIPTION
**Pull request by dlawin**
Merging all of these to a single branch/PR 

https://linear.app/datafold/issue/PX-149/cicd-with-dbt-cloud
https://linear.app/datafold/issue/PX-150/cicd-with-dbt-core-github-actions
https://linear.app/datafold/issue/PX-213/cicd-with-dbt-core-circleci
https://linear.app/datafold/issue/PX-212/cicd-with-dbt-core-gitlab

### CODE REVIEW

File: docs/guides/ci_guides/dbt_cloud.md [hunk 1]
```diff
@@ -1,4 +1,152 @@
...
+|Sync metadata on every push to prod|When selected, will sync the metadata from the dbt run with Datafold every time a push happens on the default branch.|
+|Files to ignore|If defined, the files matching the pattern will be ignored in the PRs. The pattern uses the syntax of .gitignore. Excluded files can be re-included by using the negation; re-included files can be later re-excluded again to narrow down the filter. For example, to exclude everything except the `/dbt` folder, but not the dbt `.md` files, do: <br/>`!dbt/`<br/>`dbt/*.md`|
+|Mark the CI check as failed on errors|If the checkbox is disabled, the errors in the CI runs will be reported back to GitHub/GitLab as successes, to keep the check "green" and not block the PR/MR. By default (enabled), the errors are reported as failures.|
+|Require the 'datafold' label to start CI|When this is ticked, the Datafold CI process will only run when the 'datafold' label has been applied. This label needs to be created manually in GitHub or GitLab and the title or name must match 'datafold' exactly.|
+|Sampling tolerance|The tolerance to apply in sampling for all datadiffs|
+|Sampling confidence|The confidence to apply when sampling|
```
Comment (dlawin) [inline docs/guides/ci_guides/dbt_cloud.md:132]: this seems better suited for docs/integrations

File: docs/guides/ci_guides/dbt_core/gitlab_ci.md [hunk 2]
```diff
@@ -1,4 +1,87 @@
...
+        TYPE: "production"
+        SNOWFLAKE_SCHEMA: "BEERS"
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+      variables:
+        TYPE: "pull_request"
+        SNOWFLAKE_SCHEMA: "BEERS_DEV" 
+```
+Assuming environment variables are set, may want to call out 
+
+TODO check on `check-post-upload-dbt-artifacts`
```
Comment (dlawin) [inline docs/guides/ci_guides/dbt_core/gitlab_ci.md:87]: TODO I have a slack thread open on this

Comment (dlawin) [reply docs/guides/ci_guides/dbt_core/gitlab_ci.md:87]: leave out for now

File: docs/guides/ci_guides/dbt_cloud.md (hunk 1 above)
Comment (dlawin) [reply docs/guides/ci_guides/dbt_cloud.md:132]: leaving as is

File: docs/guides/ci_guides/dbt_cloud.md [hunk 3]
```diff
@@ -1,4 +1,152 @@
...
----
\ No newline at end of file
+---
+* [Prerequisites](dbt_cloud.md#prerequisites)
+* [Basic Config](dbt_cloud.md#basic-config)
+    * [Scheduled Production Job](dbt_cloud.md#scheduled-production-job)
+    * [Merge Trigger Production Job](dbt_cloud.md#merge-trigger-production-job)
+    * [Pull Request Job](dbt_cloud.md#pull-request-job)
+* [Datafold Config](dbt_cloud.md#datafold-config)
+* [Advanced Config](dbt_cloud.md#advanced-config)
```
Comment (dlawin) [inline docs/guides/ci_guides/dbt_cloud.md:11]: Not sure if I am a fan of Basic  = no datafold, Advanced = with datafold

Maybe Base, Optimal?

File: docs/guides/ci_guides/dbt_core.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
 ---
 sidebar_position: 2
 title: dbt Core
----
\ No newline at end of file
+---
```
Comment (dlawin) [inline docs/guides/ci_guides/dbt_core.md:4]: What might we want here, if anything? @williebsweet

File: docs/guides/ci_guides/dbt_cloud.md [hunk 5]
```diff
@@ -1,4 +1,152 @@
...
+
+
+* Navigate to Jobs > Settings > Triggers > Webhooks
+* Check "Run on Pull Requests?"
+
+![](../../../static/img/cloud_pull_request_trigger.png)
+
+## Datafold Config
+
+* Navigate to Datafold > Admin > Settings > Integrations > Orchestration > Add new integration
```
[the remark this answers is not in the document]
Comment (dlawin) [reply docs/guides/ci_guides/dbt_cloud.md]: updated

File: docs/guides/ci_guides/dbt_core.md (hunk 4 above)
Comment (dlawin) [reply docs/guides/ci_guides/dbt_core.md:4]: added some cards

File: docs/guides/ci_guides/dbt_cloud.md (hunk 3 above)
Comment (dlawin) [reply docs/guides/ci_guides/dbt_cloud.md:11]: we can iterate on this, I'm gonna resolve

**Review by kylemcnair — APPROVED** [review]

**Pull request merged**